### PR TITLE
[JUP-427] add type declaration file

### PR DESCRIPTION
**Pull request by adrian-gierakowski**

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -0,0 +1,35 @@
+type loggerMethod1 = (x: string | Error) => void
+
+type LaziFy<T> = T | (() => T)
+
+type loggerMethod1Lazy = (
+  lazyX: LaziFy<Parameters<loggerMethod1>[0]>,
+) => void
+
+// TODO: be more specific about y
```
Comment (arijoon) [inline index.d.ts:9]: `y` can also contain an `error: Error` property which will be treated similar to method1, I wonder if we can capture that in the type

**Review by arijoon — APPROVED** [review]

File: index.d.ts (hunk 1 above)
Comment (adrian-gierakowski) [reply index.d.ts:9]: Would you be able to propose a more specific type for y?

Comment (adrian-gierakowski) [reply index.d.ts:9]: I'm happy to leave it for another task. I'd like to rewrite the logger in typescript (created a task for it), at which point we can define a strict interface.

Comment (arijoon) [reply index.d.ts:9]: No its not possible to have anything more specific as `y` can be anything. I was only wondering if we can somehow document that a specific property means something, e.g.:

```ts
type y = { error: Error } & Record<string, any>
```

But I think the second part will encompass the more specific error component so it won't be seen in the IDE

Comment (adrian-gierakowski) [reply index.d.ts:9]: ok, let's address this in https://darcs.atlassian.net/browse/JUP-428 by making the interface\implementation stricter

Comment (adrian-gierakowski) [reply index.d.ts:9]: I expanded the spec for the above ticket with a proposal of strict interface for the logging functions

**Pull request merged**